geode-wan/src/main/java/org/apache/geode/internal/cache/wan/parallel/ParallelGatewaySenderImpl.java

```diff
@@ -107,6 +107,9 @@ public void stop() {
       if (ev != null && !ev.isStopped()) {
         ev.stopProcessing();
       }
+      if (ev != null && ev.getDispatcher() != null) {
```

Comment:
Is it possible to pull this check and shutdown into a method?  Looks like it's used a few times throughout the code